frontend/src/components/explorer/mapViewer/ThreeDviewer.vue

```diff
@@ -219,8 +230,10 @@ export default {
       this.$store.dispatch('maps/setCoords', payload);
     },
     resetNetwork() {
-      const viewer = document.getElementById('viewer3d');
-      viewer.innerHTML = '';
+      if (this.controller) {
+        this.controller.dispose();
+        this.controller = null;
```

Comment:
Isn't setting to `null` redundant?

Reply:
This probably has something to do with the [`getInverse` -> `invert` changes](https://github.com/MetabolicAtlas/3d-network-viewer/pull/33/files#diff-35463b24cc26f87cd73e22540e83a61576bb2d35b6226a88461608713f4eb477L506-R506 ). Need to take a second look, thanks!

Reply:
Similar to the problem with the `EventBus` triggering too many times, it can also lead to the `resetNetwork` being called too many times. Setting it to `null`, as well as the initial `if` condition is to prevent this problem. I understand that it's not the best solution long-term but it solves the problem with disposing the viewer. Like you mentioned it is likely that the `EventBus` would get removed completely during the data overlay change so I hope this is ok for now.

Reply:
> This probably has something to do with the [`getInverse` -> `invert` changes](https://github.com/MetabolicAtlas/3d-network-viewer/pull/33/files#diff-35463b24cc26f87cd73e22540e83a61576bb2d35b6226a88461608713f4eb477L506-R506). Need to take a second look, thanks!

I have made a [PR](https://github.com/MetabolicAtlas/3d-network-viewer/pull/36) to address this problem. Once it's merged and the package is published we can update this repo.

Reply:
Sounds good. It might be useful to link this discussion too in the refactoring issue.

Reply:
> > This probably has something to do with the [`getInverse` -> `invert` changes](https://github.com/MetabolicAtlas/3d-network-viewer/pull/33/files#diff-35463b24cc26f87cd73e22540e83a61576bb2d35b6226a88461608713f4eb477L506-R506). Need to take a second look, thanks!
> 
> I have made a [PR](https://github.com/MetabolicAtlas/3d-network-viewer/pull/36) to address this problem. Once it's merged and the package is published we can update this repo.

Addressed in https://github.com/MetabolicAtlas/MetabolicAtlas/pull/666/commits/e1064fe8dec9aacef1711c1cecf232c54b6ab3c1